lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -1126,16 +1116,17 @@ private function getFromClauses() : array
 
         // Loop through all FROM clauses
         foreach ($this->sqlParts['from'] as $from) {
-            if ($from['alias'] === null || $from['alias'] === $from['table']) {
-                $tableSql = $from['table'];
+            /** @var From $from */
+            if ($from->alias === null || $from->alias === $from->table) {
```

Comment:
This logic could be moved into a `From` method (e.g. `hasAlias()`). It would ignore the non-null alias if it's the same as the table name.

Reply:
I'll keep it this way then.